src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/EditApprenticeship/EditApprenticeshipCommandHandler.cs

```
@@ -0,0 +1,143 @@
+        {
+            bool immediateUpdate = false;
+            if (command.EmployerReferenceUpdateRequired(apprenticeship, party))
+            {
+                apprenticeship.UpdateEmployerReference(command.EditApprenticeshipRequest.EmployerReference, party, command.EditApprenticeshipRequest.UserInfo);
+                immediateUpdate = true;
+            }
+            else
+            {
+                if (command.ProviderReferenceUpdateRequired(apprenticeship, party))
+                {
+                    apprenticeship.UpdateProviderReference(command.EditApprenticeshipRequest.ProviderReference, party, command.EditApprenticeshipRequest.UserInfo);
+                    immediateUpdate = true;
+                }
+
+                if (command.ULNUpdateRequired(apprenticeship, party))
+                {
+                    apprenticeship.UpdateULN(command.EditApprenticeshipRequest.ULN, party, _currentDateTime.UtcNow, command.EditApprenticeshipRequest.UserInfo);
+                    immediateUpdate = true;
+                }
+            }
+
+            return immediateUpdate;
+        }
+
+        private bool CreateIntermediateUpdate(EditApprenticeshipCommand command, Party party, Apprenticeship apprenticeship)
+        {
+            if (command.EditApprenticeshipRequest.IntermediateApprenticeshipUpdateRequired())
+            {
+                var apprenticeshipUpdate = command.MapToApprenticeshipUpdate(apprenticeship, party, _currentDateTime.UtcNow);
+                
+                if (!string.IsNullOrWhiteSpace(apprenticeshipUpdate.TrainingCode))
+                {
+                    var result = _mediator.Send(new GetTrainingProgrammeQuery
+                    {
+                        Id = apprenticeshipUpdate.TrainingCode
+                    }).Result;
+
+                    apprenticeshipUpdate.TrainingName = result?.TrainingProgramme?.Name;
```

is setting the training name to null valid? Would we not want to raise an error if GetTrainingProgrammeQuery cannot find the Training in question ?